src/encoded/static/components/target.js

```diff
@@ -13,28 +12,12 @@ class Target extends React.Component {
     render() {
         const context = this.props.context;
         const itemClass = globals.itemClass(context, 'view-detail key-value');
-        let geneLink;
-        let geneRef;
-        let baseName;
-        let sep;
 
-        if (context.organism.name === 'human') {
-            geneLink = dbxrefHref('HGNC', context.gene_name);
-        } else if (context.organism.name === 'mouse') {
-            const mgiRef = _(context.dbxref).find(ref => ref.substr(0, 4) === 'MGI:');
-            if (mgiRef) {
-                geneLink = dbxrefHref('MGI', mgiRef);
-            }
-        } else if (context.organism.name === 'dmelanogaster' || context.organism.name === 'celegans') {
-            const organismPrefix = context.organism.name === 'dmelanogaster' ? 'FBgn' : 'WBGene';
-            const baseUrl = context.organism.name === 'dmelanogaster' ? globals.dbxrefPrefixMap.FlyBase : globals.dbxrefPrefixMap.WormBase;
-            geneRef = _.find(context.dbxref, ref => ref.indexOf(organismPrefix) !== -1);
-            if (geneRef) {
-                sep = geneRef.indexOf(':') + 1;
-                baseName = geneRef.substring(sep, geneRef.length);
-                geneLink = baseUrl + baseName;
-            }
-        }
+        const geneIDs = context.genes.map(
+            gene => `GeneID:${gene.geneid}`
+        );
+
+        const dbxrefs = [].concat(...context.genes.map(gene => gene.dbxrefs));
```

Comment:
Simpler to do:

```
const dbxrefs = [...context.genes.map(gene => gene.dbxrefs)]
```

or probably just:

```
const dbxrefs = context.genes.map(gene => gene.dbxrefs)
```

...because it doesn’t appear to matter if the array is cloned or not.

Reply:
gene.dbxrefs itself is an array. I'm trying to flat the array here. I think I need a flat array [here](https://github.com/ENCODE-DCC/encoded/pull/2372/files/73d38b88d200695c4563e350a956e480a67e7194#diff-1b76489ca7832bd8baec06af55cf04a7R63). Does this sound right/reasonable?

Reply:
You’re converting an array of gene objects into a corresponding array of gene dbxrefs?

Reply:
Oh wait, I have to think about this. You want a simple array of dbxref strings that combines all the arrays in all genes?

Reply:
I think that’s what you’re doing, right? In that case, I think this is fine but could you add a comment so it’s clear what this is for?

Reply:
Sure. Will add a comment.

To clarify, target.genes is an array of gene objects. For a gene object, gene.dbxrefs is an array of dbxrefs (strings). This PR will remove the dbxref property from target object. To display dbxrefs related to a target, using/extracting dbxrefs from its related/targeted genes. So yes, it will be a a simple array of dbxref strings that combines all the (gene.dbxrefs) arrays in all genes. Is this making sense now?